lib/evaluate/fold.cc

```diff
@@ -473,14 +474,48 @@ Expr<Type<TypeCategory::Integer, KIND>> FoldOperation(FoldingContext &context,
       }
       return FoldElementalIntrinsic<T, T, T, T>(
           context, std::move(funcRef), &Scalar<T>::MERGE_BITS);
+    } else if (name == "rank") {
+      // TODO assumed-rank dummy argument
+      return Expr<T>{args[0].value().Rank()};
+    } else if (name == "shape") {
+      if (auto shape{GetShape(args[0].value())}) {
+        if (auto shapeExpr{AsShapeArrayExpr(*shape)}) {
+          return Fold(context, ConvertToType<T>(std::move(*shapeExpr)));
+        }
+      }
+    } else if (name == "size") {
+      if (auto shape{GetShape(args[0].value())}) {
+        if (auto &dimArg{args[1]}) {  // DIM= is present, get one extent
+          if (auto dim{ToInt64(dimArg->value())}) {
+            std::int64_t rank = shape->size();
```

Comment:
Use brace init here?

Reply:
The result type of `size()` is unsigned and brace initialization elicits a warning about possible truncation.